### PR TITLE
fix: lists and list item blocks in WP 6.1

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

WP 6.1 introduced the concept of sortable list items. This, however, broke our MJML renderer's implementation of lists, causing them to not be rendered at all in the final email HTML. The fact that list items are now inner blocks of List blocks also means that we need to whitelist `core/list-item` as an allowed block in the Newsletter editor, or it's not possible to add list items to any list.

### How to test the changes in this Pull Request:

1. On a site with WP 6.1, on `master`, create a newsletter containing two List blocks: one bulleted and the other numbered. Try to add some list items and observe that you can't.
2. Check out this branch and refresh the editor. Confirm that you can now add list items to each List block.
3. Send a test email. Confirm that the lists are rendered as expected.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by leogermani — APPROVED** [review]
I was able to test everything but the actual sent email - probably because of something on my setup here.

Maybe it's worth someone else double checking it

**Comment by dkoo** [comment]
Thanks, @leogermani! I'll wait for another approval from @Automattic/newspack-product before merging.

**Pull request merged**
